tests/Unit/ApparentHorizons/Test_Strahlkorper.cpp

```diff
@@ -0,0 +1,457 @@
+  StrahlkorperDB::types<Frame::Inertial>::OneForm test_dx_radius(n_pts);
+  for (size_t s = 0; s < n_pts; ++s) {
+    // Analytic solution I computed in Mathematica
+    const double theta = theta_phi[0][s];
+    const double phi = theta_phi[1][s];
+    const double r = test_radius[s];
+    const double sin_phi = sin(phi);
+    const double cos_phi = cos(phi);
+    const double sin_theta = sin(theta);
+    const double cos_theta = cos(theta);
+    const double amp = -sqrt(3.0 / 8.0 / M_PI) * y11_amplitude;
+
+    test_dx_radius.get(0)[s] = -((cos_phi * sin_phi) / r) +
+                               (cos_phi * square(cos_theta) * sin_phi) / r;
+    test_dx_radius.get(1)[s] =
+        square(cos_phi) / r + (square(cos_theta) * square(sin_phi)) / r;
+    test_dx_radius.get(2)[s] = -((cos_theta * sin_phi * sin_theta) / r);
+    for (auto& a : test_dx_radius) {
+      a[s] *= amp;
+    }
+  }
+  const auto& strahlkorper_dx_radius =
+      db::get<StrahlkorperDB::DxRadius<Frame::Inertial>>(box);
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t s = 0; s < n_pts; ++s) {
+      CHECK(strahlkorper_dx_radius.get(i)[s] ==
+            approx(test_dx_radius.get(i)[s]));
+    }
+  }
+
+  // Test second derivatives.
+  StrahlkorperDB::types<Frame::Inertial>::SecondDeriv test_d2x_radius(n_pts);
+  for (size_t s = 0; s < n_pts; ++s) {
+    // Messy analytic solution I computed in Mathematica
+    const double theta = theta_phi[0][s];
+    const double phi = theta_phi[1][s];
+    const double r = test_radius[s];
+    const double sin_theta = sin(theta);
+    const double cos_theta = cos(theta);
+    const double sin_phi = sin(phi);
+    const double cos_phi = cos(phi);
+    const double cos_2_theta = cos(2 * theta);
+    const double amp = -sqrt(3.0 / 8.0 / M_PI) * y11_amplitude;
+    test_d2x_radius.get(0, 0)[s] =
+        (9 * sin(3 * phi) * sin_theta -
+         sin_phi * (7 * sin_theta + 12 * square(cos_phi) * sin(3 * theta))) /
+        (16. * square(r));
+    test_d2x_radius.get(0, 1)[s] =
+        -((cos_phi *
+           (square(cos_phi) + ((-1 + 3 * cos_2_theta) * square(sin_phi)) / 2.) *
+           sin_theta) /
+          square(r));
+    test_d2x_radius.get(0, 2)[s] =
+        (3 * cos_phi * cos_theta * sin_phi * square(sin_theta)) / square(r);
+    test_d2x_radius.get(1, 1)[s] =
+        (-3 * (7 * sin_phi * sin_theta + 3 * sin(3 * phi) * sin_theta +
+               4 * pow<3>(sin_phi) * sin(3 * theta))) /
+        (16. * square(r));
+    test_d2x_radius.get(1, 2)[s] =
+        -((cos_theta * (square(cos_phi) +
+                        ((-1 + 3 * cos_2_theta) * square(sin_phi)) / 2.)) /
+          square(r));
+    test_d2x_radius.get(2, 2)[s] =
+        ((1 + 3 * cos_2_theta) * sin_phi * sin_theta) / (2. * square(r));
+    for (auto& a : test_d2x_radius) {
+      a[s] *= amp;
+    }
+  }
+  const auto& strahlkorper_d2x_radius =
+      db::get<StrahlkorperDB::D2xRadius<Frame::Inertial>>(box);
+  for (size_t i = 0; i < 3; ++i) {
```

Comment:
try CIA

Reply:
Again since the Tensors are iterable it should possible to eliminate the outer loops